docs/DeveloperGuide.md

```diff
@@ -276,6 +276,65 @@ As a result, editing a supplier's details necessitate a corresponding update to
 
 <img src="images/SupplierAndSupplyItemClassDiagram.png" width="500" />
 
+### \[Developed\] AddTask feature
+
+**Implementation**
+
+This feature allow us to `Tasks` to be added to the `TaskList` for vendors to keep track of tasks they need to do.
+
+The proposed addTask feature is facilitated by `AddTaskCommand` and `AddTaskCommandParser` classes
+
+The `AddTaskCommandParser` parses the input given by the user and ensures that all input required to create the
+tasks are present and valid. Once inputs are parsed and are valid, a `Task` instance will be created and will be
+added to the taskList.
+
+The feature will be implemented with the help of the following operations:
+* `Model#addTask(Task task)` which adds a newly created `Task` instance to the taskList.
+* `AddTaskCommandParser#parse(String arg)` which will parse the input then create a new task then returns an 
+AddTaskCommand with the new task created.
+
+Given below is an example of how `AddTaskCommand` is being executed
+
+**Steps**
+
+Step 1. The user enters the `addTask d/Buy Chicken dl/2020-12-12 t/Food` command.
+d/ represents the task description, dl/ represents deadline and t/ represents tag.
+
+Step 2. The `AddTaskCommandParser` parses the input and ensures that command is valid.
+For it to be valid, all compulsory field such as deadline and descriptions must be present.
+Tag field is optional. Further checks are that date given are in the correct format of 
+`yyyy-mm-dd`. If these checks are not valid, an exception will be thrown.
```

Comment:
First check is for whether is yyyy-mm-dd.
Second check is for whether the yyyy, mm or dd is valid as per Georgian calender, i.e. months must be 1 <= month <= 12 and day is 1-28/29/30/31 (Feb esp)

Reply:
Noted, I have added the more specific check to the developer's guide